src/istgt_iscsi.c

```diff
@@ -7180,6 +7203,42 @@ istgt_iscsi_drop_all_conns(CONN_Ptr conn)
 	return (0);
 }
 
+// istgt_iscsi_is_existing_conn_alive returns 1 if
+// there are any existing connections that are alive
+// else return 0
+static int
+istgt_iscsi_is_existing_conn_alive(CONN_Ptr conn)
+{
+	CONN_Ptr existing_conn = NULL;
+	int i;
+
+	MTX_LOCK(&g_conns_mutex);
+	for (i = 0; i <= g_max_connidx; i++) {
+		existing_conn = g_conns[i];
+		if (existing_conn == NULL)
+			continue;
+		if (existing_conn == conn)
+			continue;
+		// If new connection is totaly different from existing connection
+		// then check the status of existing connection if it is active
+		// return true
+		if (strcasecmp(conn->initiator_port, existing_conn->initiator_port) != 0
+			&& strcasecmp(conn->target_name, existing_conn->target_name) == 0
+			&& strcasecmp(conn->initiator_name, existing_conn->initiator_name) != 0) {
+			if (existing_conn->state == CONN_STATE_RUNNING) {
```

Comment:
what are other possible cases? and when does the state gets changed?

Reply:
Other possible states exist [here](https://github.com/openebs/istgt/blob/31d7b87a2f8d2fb0f38fbaea12904f30d00f6348/src/istgt_iscsi.h#L181-L186)... 
- Initial state of conn->state is CONN_STATE_INVALID.
- AFAIK after spinning the worker thread(which will get spanned after a client connects to target) & before spinning the sender thread we are marking conn->state to `CONN_STATE_RUNNING`.
- Following are the cases where we mark it as `CONN_STATE_EXITING`
  - When there is an error in target while reading data from an initiator.
  - When the target is not ready to accept IOs.
  - A case where executing opcode got failed.